### PR TITLE
Set gp3 as the default VolumeType

### PR DESCRIPTION
**Pull request by Callisto13**
Closes https://github.com/weaveworks/eksctl/issues/2891.

### Description

- Set gp3 as the default volumeType
- Provide default settings for IOPS and Throughput for gp3 type
- Provide default setting for IOPS io1 type
- Ensure given IOPS and Throughput values are within accepted ranges

### Checklist
- [x] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`), target version (e.g. `version/0.12.0`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [x] Backfilled missing tests for code in same general area :tada:
- [x] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/apis/eksctl.io/v1alpha5/types.go [hunk 1]
```diff
@@ -279,7 +288,7 @@ var (
 	DefaultNodeSSHPublicKeyPath = "~/.ssh/id_rsa.pub"
 
 	// DefaultNodeVolumeType defines the default root volume type to use
-	DefaultNodeVolumeType = NodeVolumeTypeGP2
+	DefaultNodeVolumeType = NodeVolumeTypeGP3
```
Comment (aclevername) [inline pkg/apis/eksctl.io/v1alpha5/types.go:291]: does changing the default effect the way we interactive with existing nodegrooups at all? I imagine this wasn't being explicitly set most if the time

File: pkg/apis/eksctl.io/v1alpha5/types.go [hunk 2]
```diff
@@ -269,6 +272,12 @@ const (
 	NodeGroupTypeUnmanaged NodeGroupType = "unmanaged"
 	// NodeGroupTypeUnowned defines an unowned managed nodegroup
 	NodeGroupTypeUnowned NodeGroupType = "unowned"
+	// DefaultNodeVolumeThroughput defines the default throughput for gp3 volumes
+	DefaultNodeVolumeThroughput = 125
+	// DefaultNodeVolumeIO1IOPS defines the default throughput for gp3 and io1 volumes
+	DefaultNodeVolumeIO1IOPS = 100
+	// DefaultNodeVolumeGP3IOPS defines the default throughput for gp3 and io1 volumes
+	DefaultNodeVolumeGP3IOPS = 3000
```
Comment (aclevername) [inline pkg/apis/eksctl.io/v1alpha5/types.go:280]: Are these defaults from an AWS doc? Could we link it in the comment and/or in the PR?

File: pkg/apis/eksctl.io/v1alpha5/types.go [hunk 3]
```diff
@@ -249,8 +249,11 @@ const (
 
 // Values for `VolumeType`
 const (
-	// NodeVolumeTypeGP2 is General Purpose SSD (default)
+	// NodeVolumeTypeGP2 is General Purpose SSD
 	NodeVolumeTypeGP2 = "gp2"
+	// NodeVolumeTypeGP3 is General Purpose SSD which can be optimised for
```
Comment (michaelbeaumont) [inline pkg/apis/eksctl.io/v1alpha5/types.go:254]: ah looks like the json schema generator doesn't like newlines in these comments atm

**Comment by Callisto13** [comment]
woah who merged this? it was not done

**Comment by michaelbeaumont** [comment]
I merged it, you can see I added the autosquash label.

**Comment by Callisto13** [comment]
👍  yep i saw once i had woken up a bit. I'll open a new PR to handle the rest

**Comment by michaelbeaumont** [comment]
Sorry, it seemed rather complete. What's missing?

**Comment by Callisto13** [comment]
When manual testing I realised it was 100% working for managed ngs

**Pull request merged**